test/contracts/wormholeJoin.ts

```diff
@@ -0,0 +1,137 @@
+  const constantFee = await ConstantFeeFactory.deploy(globalFee, globalFeeTTL)
+  console.log('ConstantFee deployed at: ', constantFee.address)
+
+  const WormholeOracleAuthFactory = getContractFactory<WormholeOracleAuth__factory>('WormholeOracleAuth', defaultSigner)
+  const oracleAuth = await WormholeOracleAuthFactory.deploy(join.address)
+
+  console.log('Deploying router...')
+  const WormholeRouterFactory = getContractFactory<WormholeRouter__factory>('WormholeRouter', defaultSigner)
+  const router = await WormholeRouterFactory.deploy(sdk.dai.address)
+
+  console.log('Finalizing permissions...')
+  await waitForTx(join.rely(oracleAuth.address))
+  await waitForTx(join.rely(router.address))
+  await waitForTx(join.rely(sdk.pause_proxy.address))
+  await waitForTx(join.deny(await defaultSigner.getAddress()))
+
+  await waitForTx(oracleAuth.rely(sdk.pause_proxy.address))
+  await waitForTx(oracleAuth.deny(await defaultSigner.getAddress()))
+
+  await waitForTx(router.rely(sdk.pause_proxy.address))
+  await waitForTx(router.deny(await defaultSigner.getAddress()))
+
+  return { join, oracleAuth, router, constantFee }
+}
+export type WormholeSdk = Awaited<ReturnType<typeof deployWormhole>>
+
+export async function configureWormhole({
+  sdk,
+  wormholeSdk,
+  joinDomain,
+  defaultSigner,
+  domainsCfg,
+  oracleAddresses,
+  globalLine,
+  baseBridgeSdk,
+}: {
+  defaultSigner: Signer
+  globalLine: BigNumber
+  domainsCfg: Dictionary<{ line: BigNumber; l1Bridge: string }>
+  joinDomain: string
+  oracleAddresses: string[]
+  sdk: MainnetSdk
+  wormholeSdk: WormholeSdk
+  baseBridgeSdk: BaseBridgeSdk
+}) {
+  const L1ConfigureWormholeSpellFactory = getContractFactory<L1ConfigureWormholeSpell__factory>(
+    'L1ConfigureWormholeSpell',
+    defaultSigner,
+  )
+  const configureSpell = await L1ConfigureWormholeSpellFactory.deploy(
+    joinDomain,
+    wormholeSdk.join.address,
+    sdk.vow.address,
+    sdk.vat.address,
+    globalLine,
+    wormholeSdk.router.address,
+  )
+
+  await executeSpell(sdk, configureSpell)
+
+  for (const [domainName, domainCfg] of Object.entries(domainsCfg)) {
+    const L1AddWormholeDomainSpellFactory = getContractFactory<L1AddWormholeDomainSpell__factory>(
+      'L1AddWormholeDomainSpell',
+      defaultSigner,
+    )
+    const addWormholeDomainSpell = await L1AddWormholeDomainSpellFactory.deploy(
+      domainName,
+      wormholeSdk.join.address,
+      wormholeSdk.constantFee.address,
+      domainCfg.line,
+      wormholeSdk.router.address,
+      domainCfg.l1Bridge,
+      baseBridgeSdk.l1Escrow.address,
+      sdk.dai.address,
+    )
+    console.log('Executing spell to add a new domain...')
+    await executeSpell(sdk, addWormholeDomainSpell)
+  }
+
+  console.log('Configuring oracleAuth...')
+  const govImpersonator = await impersonateAccount(sdk.pause_proxy.address, sdk.dai.provider as any)
+  await wormholeSdk.oracleAuth
```

Comment:
Why aren't the 2 following lines included in `L1ConfigureWormholeSpell.sol` ?

Reply:
This is quite unfortunate but I couldn't add an array of arguments (list of oracles) to spell b/c it has to be immutable (spell is delegate called) and you can't have immutable arrays in solidity. But when i think about it I will just assume there's always 3 oracles in tests 👍